1-js/06-advanced-functions/03-closure/10-make-army/solution.md

```diff
@@ -51,24 +51,29 @@ function makeArmy() {
 }
 ```
 
-...We can see that it lives in the lexical environment associated with the current `makeArmy()` run. But when `army[5]()` is called, `makeArmy` has already finished its job, and `i` has the last value: `10` (the end of `while`).
+ we can see that it lives in the lexical environment associated with the current `makeArmy()` run. But when `army[5]()` is called, `makeArmy` has already finished its job, and the final value of `i` is `10` (at the end of `while`).
 
-As a result, all `shooter` functions get from the outer lexical envrironment the same, last value `i=10`.
+As a result, all `shooter` functions get the same value from the outer lexical environment and that is, the last value, `i=10`.
 
-We can fix it by moving the variable definition into the loop:
+![](task-while-lexenv-makearmy.svg)
```

Comment:
Is that your new picture?

Reply:
Yes. Is there any problem with it?

Reply:
It's a new one. Do you use Sketch(Mac) for creating SVGs?

Reply:
Yes, I'll import it myself.